Neos.Flow/Classes/ObjectManagement/DependencyInjection/ProxyClassBuilder.php

```diff
@@ -466,7 +468,33 @@ public function buildPropertyInjectionCodeByString(Configuration $objectConfigur
             return $result;
         }
 
-        if ($propertyConfiguration->isLazyLoading() && $this->objectConfigurations[$propertyObjectName]->getScope() !== Configuration::SCOPE_PROTOTYPE) {
+        $propertyClassHasInterfaceWithConstructor = false;
+        $propertyClassHasFinalMethod = false;
+        if ($propertyClassName) {
+            foreach ((new ClassReflection($propertyClassName))->getInterfaceNames() as $interfaceName) {
+                if (method_exists($interfaceName, '__construct')) {
+                    $propertyClassHasInterfaceWithConstructor = true;
+                    break;
+                }
+            }
+            foreach (get_class_methods($propertyClassName) as $methodName) {
+                if ($this->reflectionService->isMethodFinal($propertyClassName, $methodName)) {
+                    $propertyClassHasFinalMethod = true;
+                }
+            }
+        }
+
+        # There are several cases when providing lazy loading is not possible, due to the way
+        # lazy loading is implemented. If any of the check fails, we fall back to eager loading:
+        if (
+            $propertyConfiguration->isLazyLoading() &&
+            $this->objectConfigurations[$propertyObjectName]->getScope() !== Configuration::SCOPE_PROTOTYPE &&
+            !interface_exists($this->objectConfigurations[$propertyObjectName]->getClassName()) &&
```

Comment:
If the target class is an interface, but the interface is Objects.yaml configured to a specific implementation, why wouldn't that work lazily?

Reply:
that doesnt seem to be resolved yet - or am i wrong?
we say in the docs that this would work - and codewise, since that `interface_exists` was just added it must have worked ... https://flowframework.readthedocs.io/en/stable/TheDefinitiveGuide/PartIII/ObjectManagement.html?highlight=lazy#lazy-dependency-injection

can we still support this?